app/models.py

```diff
@@ -220,7 +224,11 @@ def __init__(
             raise InventoryException(title="Invalid request", detail="The tags field cannot be null.")
 
         super().__init__(canonical_facts, display_name, ansible_host, account, facts, tags, system_profile_facts)
-        self.stale_timestamp = stale_timestamp
+
+        if system_profile_facts and system_profile_facts.get("host_type") == "edge":
```

Comment:
I ran into this concept recently and it seemed neat, so I thought i'd suggest it. You can do JS style optional chaining in python pretty easily and it simplifies checks like this nicely. This one is only two checks so it's far from worst case but if you want to try the alternative I've suggested the changes below. The behavior is identical either way.

I realized after writing that that in this case we don't know if the root object is None or not, so it doesn't come out looking quite as nice as when you know it exists. Like with `self` which looks like this.

```
self.get(system_profile_facts, {}).get("host_type") == "edge"
```

anyways, take it or leave it, just thought it was interesting, and I always hate writing checks like this.

```suggestion
        if (system_profile_facts or {}).get("host_type") == "edge":
```

Reply:
Plan to leave it this time.

Reply:
I'm fine with either implementation. I do prefer the more concise one and didn't realize you could do that in Python, so thanks for the info!